Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean

```diff
@@ -927,6 +937,77 @@ def localizationMapOfSpecializes {x y : PrimeSpectrum R} (h : x ⤳ y) :
+  map_rel_iff' {p q} :=
+    (RelIso.symm irreducibleSetEquivPoints).map_rel_iff.trans (le_iff_specializes p q).symm
+
+end PrimeSpectrum
+
+open PrimeSpectrum in
+/--
+[Stacks: Lemma 00ES (3)](https://stacks.math.columbia.edu/tag/00ES)
+Zero loci of minimal prime ideals of `R` are irreducible components in `Spec R` and any
+irreducible component is a zero locus of some minimal prime ideal.
+-/
+protected def minimalPrimes.equivIrreducibleComponents :
+    minimalPrimes R ≃o (irreducibleComponents <| PrimeSpectrum R)ᵒᵈ :=
+  let e : {p : Ideal R | p.IsPrime ∧ ⊥ ≤ p} ≃o PrimeSpectrum R :=
+    ⟨⟨fun x ↦ ⟨x.1, x.2.1⟩, fun x ↦ ⟨x.1, x.2, bot_le⟩, fun _ ↦ rfl, fun _ ↦ rfl⟩, Iff.rfl⟩
+  (e.trans <| PrimeSpectrum.pointsEquivIrreducibleCloseds R).minimalsIsoMaximals.trans
+    (OrderIso.setCongr _ _ <| by simp_rw [irreducibleComponents_eq_maximals_closed, and_comm]).dual
+
+open PrimeSpectrum in
+lemma minimalPrimes.eq_irreducibleComponents :
+    minimalPrimes R =
+    PrimeSpectrum.vanishingIdeal '' (irreducibleComponents <| PrimeSpectrum R) := by
+  rw [irreducibleComponents_eq_maximals_closed, minimalPrimes_eq_minimals, ← minimals_swap,
+    ← PrimeSpectrum.vanishingIdeal_isClosed_isIrreducible, image_minimals_of_rel_iff_rel]
+  intro s t hs _
+  exact vanishingIdeal_anti_mono_iff hs.1
+
+namespace PrimeSpectrum
+
+lemma zeroLocus_minimalPrimes :
+    zeroLocus ∘ (↑) '' minimalPrimes R =
+    irreducibleComponents (PrimeSpectrum R) := by
+  rw [minimalPrimes.eq_irreducibleComponents, ← Set.image_comp, Set.EqOn.image_eq_self]
+  intros s hs
+  simpa [zeroLocus_vanishingIdeal_eq_closure, closure_eq_iff_isClosed]
+    using isClosed_of_mem_irreducibleComponents s hs
+
+lemma mem_irreducibleComponents_iff_vanishingIdeal_mem_minimalPrimes
+    {s : Set (PrimeSpectrum R)} (hs : IsClosed s) :
+    s ∈ irreducibleComponents (PrimeSpectrum R) ↔ (vanishingIdeal s) ∈ minimalPrimes R := by
+  constructor
+  · rw [minimalPrimes.eq_irreducibleComponents]; exact Set.mem_image_of_mem _
+  · conv_rhs => rw [← zeroLocus_minimalPrimes, ← hs.closure_eq,
+      ← zeroLocus_vanishingIdeal_eq_closure]
+    exact Set.mem_image_of_mem _
+
+lemma zeroLocus_ideal_mem_irreducibleComponents_iff_mem_minimalPrimes (I : Ideal R) :
+    zeroLocus I ∈ irreducibleComponents (PrimeSpectrum R) ↔
+    ∃ (J : Ideal R), J ∈ minimalPrimes R ∧ zeroLocus I = zeroLocus (J : Set R) := by
+  rw [← PrimeSpectrum.zeroLocus_minimalPrimes]
+  simp_rw [eq_comm (a := zeroLocus (I : Set R))]
+  rfl
+
+lemma zeroLocus_ideal_mem_irreducibleComponents_iff_mem_minimalPrimes_of_isRadical
+    {I : Ideal R} (hI : I.IsRadical) :
+    zeroLocus I ∈ irreducibleComponents (PrimeSpectrum R) ↔ I ∈ minimalPrimes R := by
+  constructor
+  · conv_rhs => rw [minimalPrimes.eq_irreducibleComponents, ← hI.radical,
+      ← vanishingIdeal_zeroLocus_eq_radical]
+    exact Set.mem_image_of_mem _
+  · rw [← PrimeSpectrum.zeroLocus_minimalPrimes]
+    exact Set.mem_image_of_mem _
+
```

Comment:
I think these should be removed in favor of 
```lean
lemma vanishingIdeal_mem_minimalPrimes
    {s : Set (PrimeSpectrum R)} :
    (vanishingIdeal s) ∈ minimalPrimes R ↔ closure s ∈ irreducibleComponents (PrimeSpectrum R) := by
  constructor
  · conv_rhs => rw [← zeroLocus_minimalPrimes,
      ← zeroLocus_vanishingIdeal_eq_closure]
    exact Set.mem_image_of_mem _
  · rw [minimalPrimes.eq_irreducibleComponents, ← vanishingIdeal_closure]
    exact Set.mem_image_of_mem _

lemma zeroLocus_ideal_mem_irreducibleComponents
    {I : Ideal R} :
    zeroLocus I ∈ irreducibleComponents (PrimeSpectrum R) ↔ I.radical ∈ minimalPrimes R := by
  constructor
  · conv_rhs => rw [minimalPrimes.eq_irreducibleComponents,
      ← vanishingIdeal_zeroLocus_eq_radical]
    exact Set.mem_image_of_mem _
  · rw [← PrimeSpectrum.zeroLocus_minimalPrimes, ← zeroLocus_radical]
    exact Set.mem_image_of_mem _
```
maybe even simp lemmas?

Reply:
They get rid of `vanishingIdeal` and `zeroLocus` though. Anyway I don't have a strong preference and I'm fine with either way.